Modules/Core/Support/MailerHelper.php

```
@@ -93,4 +93,139 @@ public static function check_mail_errors(array $errors = [], $redirect = ''): vo
+
+        if ($bcc && ! self::validate_email_address($bcc)) {
+            $errors[] = 'bcc_email';
+        }
+
+        self::check_mail_errors($errors, 'mailer/invoice/' . $invoice_id);
+
+        $message = (empty($message) ? ' ' : $message);
+
+        // Use the namespaced phpmail_send function
+        return \Modules\Sessions\Controllers\phpmail_send($from, $to, $subject, $message, $invoice, $cc, $bcc, $attachments);
+    }
```

PHPMD (2.15.0)

The method email_invoice() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)